packages/fx-core/src/plugins/resource/apiconnector/plugin.ts

```diff
@@ -112,21 +123,27 @@ export class ApiConnectorImpl {
           );
         })
       );
-      if (err instanceof SystemError || err instanceof UserError) {
-        throw err;
-      } else {
-        throw ResultFactory.SystemError(
+      if (!(err instanceof SystemError) && !(err instanceof UserError)) {
+        err = ResultFactory.SystemError(
           ErrorMessage.generateApiConFilesError.name,
           ErrorMessage.generateApiConFilesError.message(err.message)
         );
       }
+      const thrownErr = err as FxError;
```

Comment:
can we extract line 132 -136 as a util function?

Reply:
It is fixed. Thanks.